src/v/raft/replicate_entries_stm.cc

```diff
@@ -230,8 +231,9 @@ replicate_entries_stm::apply(std::vector<ss::semaphore_units<>> u) {
           });
           // Wait until all RPCs will be dispatched
           return _dispatch_sem.wait(requests_count)
-            .then([this, append_result, u = std::move(u)]() mutable {
+            .then([this, append_result]() mutable {
                 _req.reset();
+                _units.release();
```

Comment:
it looks like the lifetime of `u` before and `_units` now are the same... am I missing something?